tools/autograd/templates/variable_factories.h

```diff
@@ -186,12 +186,18 @@ AT_FORALL_SCALAR_TYPES_AND3(Bool, Half, BFloat16, TENSOR)
 /// support it in the future by iterating over all sub-lists to find
 /// the largest data type that can represent all of the elements, or by using
 /// variadic templates.
-inline at::Tensor tensor(detail::InitListTensor list_init_tensor, const at::TensorOptions& options) {
-  return autograd::make_variable(list_init_tensor.to_tensor(options), options.requires_grad());
+inline at::Tensor tensor(detail::InitListTensor init_list_tensor, const at::TensorOptions& options) {
+  return autograd::make_variable(init_list_tensor.to_tensor(options), options.requires_grad());
 }
 
-inline at::Tensor tensor(detail::InitListTensor list_init_tensor) {
-  return torch::tensor(list_init_tensor, at::dtype(list_init_tensor.scalar_type()));
+inline at::Tensor tensor(detail::InitListTensor init_list_tensor) {
+  return torch::tensor(init_list_tensor, at::dtype(init_list_tensor.scalar_type()));
+}
+
+/// NOTE: We add `torch::tensor(std::initializer_list<detail::InitListTensor>)` function overload,
+/// so that `torch::tensor({{1, 2}})` can take this overload instead of `torch::tensor(at::ArrayRef<T>)`.
+inline at::Tensor tensor(std::initializer_list<detail::InitListTensor> init_list) {
+  return torch::tensor(detail::InitListTensor(init_list), at::dtype(init_list.begin()->scalar_type()));
```

Comment:
I ran a test and can confirm that adding this doesn't cause `torch::tensor({1, 2, 3})` to take the slower path.

Reply:
TODO: add TORCH_CHECK to check that the slower path is never taken for `torch::tensor({1, 2, 3})`